src/chapters/classes/constructors.rst

```diff
@@ -0,0 +1,157 @@
+number, order, and type of the constructor arguments.
+
+Let’s expand upon our ``Student`` class.
+
+.. admonition:: Example
+
+   .. sourcecode:: java
+      :linenos:
+
+      public class Student {
+
+         private String name;
+         private final int studentId;
+         private int numberOfCredits;
+         private double gpa;
+
+         public Student(String name, int studentId,
+                  int numberOfCredits, double gpa) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = numberOfCredits;
+            this.gpa = gpa;
+         }
+
+         public Student(String name, int studentId) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = 0;
+            this.gpa = 0.0;
+         }
+
+         /* getters and setters omitted */
+
+      }
+
+   The first constructor allows for the creation of ``Student`` objects
+   where the code creating the object provides initial values for each of
+   the fields. The second allows for the creation of ``Student`` objects
+   with only ``name`` and ``studentId``. The first constructor would be
+   most useful for creating a transfer student, where credits and a GPA
+   might already be non-zero. However, for all new students, it would be
+   safe to initialize ``numberOfCredits`` and ``gpa`` to be 0.
+
+   A better way to write the above constructors would be this:
+
+   .. sourcecode:: java
+      :linenos:
+
+      public class Student {
+
+         private String name;
+         private final int studentId;
+         private int numberOfCredits;
+         private double gpa;
+
+         public Student(String name, int studentId,
+                  int numberOfCredits, double gpa) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = numberOfCredits;
+            this.gpa = gpa;
+         }
+
+         public Student(String name, int studentId) {
+            this(name, studentId, 0, 0);
+         }
+
+         /* getters and setters omitted */
+
+      }
+
+   Here, we use ``this()`` to invoke another constructor within the same
+   class. In this case, the second constructor calls the first with the
+   “default” values for ``numberOfCredits`` and ``gpa``. If you use this
+   syntax, the call to ``this()`` must be the first line in the
+   constructor. This is a good practice not only because it makes your code
+   shorter, but also because it allows any initialization behavior that may
+   be carried out beyond just initializing variables to be contained in a
+   smaller number of constructors. In other words, constructors can “share”
+   initialization code. Notice from this example that a constructor doesn’t
+   need to require an initial value for each field as an argument.
+
+.. note::
+
+   When defining constructors, think about: 1. Which fields must be
```

Comment:
check the formatting on this list, its coming back as all 1.s and is inline text

Reply:
also, a concept check or two with practice on using a constructor or this() could be well placed here!

Reply:
Fixed! Thanks!